accepted/0000-no-package-json-_fields.md

```diff
@@ -0,0 +1,294 @@
+  dependencies, this is the tarball url that was ultimately fetched.  For
+  git dependencies, this is the requested git url with `#<commit-hash>`
+  attached to the end.  Again, this metadata is important because we must
+  ensure that we get the same package tree between runs and on different
+  machines.
+
+- `_from` - Legacy form of `_requested`.  Passed to npa to turn into a
+  parsed object.
+
+- `_spec` - Just an copy of `_requested.rawSpec`.  Unnecessary.
+
+- `_id` - `${name}@${version}`.  Unnecessary.
+
+- `_requiredBy` - This is load bearing currently, to create the package
+  trees shown in `npm ls`.  However, Arborist adds a `node.edgesIn` field
+  which lists all the dependencies on a given node along with their types.
+
+    Still, having that information in-place is useful when attempting to
+    determine why a given package in the file tree is being included.
+    Using `npm ls`, this can be difficult to determine, since `npm ls` does
+    not provide a facility for filtering by the _location_ of a node, only
+    by its _name_.
+
+    A subsequent RFC will discuss adding `npm ls <location>`, which will
+    address this use case in a more user-friendly way than looking at
+    implementation details directly.
+
+    Conclusion: can be removed.
+
+- `_shrinkwrap` - This is a complete copy of the `npm-shrinkwrap.json` file
+  if it exists in the package.  Since it's a duplicate of data that already
+  exists (in roughly the same location), it can be safely omitted.
+
+While a `remote` or `git` dependency might satisfy a version dependency
+(based on the same version number), but the reverse is not true.
+Arguably, if we have a dependency on a git or remote dep, then a
+non-git or non-remote dep is not valid.
+
+### Proposal
+
+Add a file at `node_modules/.arborist-resolutions.json` which stores the
+only two pieces of metadata that npm really needs.  This file has the
+following format:
+
+```json
+{
+  "/package-name": {
+    "resolved": "${current _resolved field}",
+    "integrity": "${current _integrity field}"
+  },
+  "/package-name/nested-dep": {
+    "resolved": "${url}",
+    "integrity": "${sri}"
+  },
+  "..."
+}
+```
+
+The key is the `Node.location` field.  Metadata that would have been
+stored in `_integrity` and `_resolved` fields in
+`node_modules/a/node_modules/b/package.json` will be in the
+`.arborist-resolutions.json` under the key `/a/b`.
+
+This information will _not_ be tracked for `bundleDependencies`, as they
+are never fetched directly.
+
+If the `node_modules/.arborist-resolutions.json` file is not present, then
+attempt to get the reso
+
+## Rationale and Alternatives
+
+This is closest to how pnpm stores its lock file in
+`node_modules/.pnpm-lock.yaml`.  However, rather than storing the entire
+lock file in `node_modules`, npm will only store the information that
+cannot be read from the reified package hierarchy if the package-lock is
+missing.
```

Comment:
Does this mean if you have a `package-lock.json`, no `node_modules/.arcanist-resolutions.json` will be created? Or am I understanding this sentence wrong?

Reply:
Ah, that is ambiguous, I'll reword it.

It means: "it will save only (the information that would be unavailable without a package-lock.json)", not "it will save the information (only without a package-lock)".

Ie, always save, but only save what can't get from node_modules itself.